File: app/src/main/java/com/mehul/redditwall/BitURL.java
package com.mehul.redditwall;

import android.graphics.Bitmap;

import com.google.gson.annotations.SerializedName;

public class BitURL {
    @SerializedName("img")
    private Bitmap img;
    @SerializedName("url")
    private String url;
    @SerializedName("gif")
    private boolean containsGif = false;

    BitURL(Bitmap i, String u) {
        img = i;
        url = u;
        if (img == null) {
            containsGif = true;
        }
    }

    Bitmap getImg() {
        return img;
    }

    String getUrl() {
        return url;
    }

    boolean hasGif() {
        return containsGif;
    }

    void setGif(boolean gif) {
        containsGif = gif;
    }
}

File: app/src/main/java/com/mehul/redditwall/BitURL.kt
package com.mehul.redditwall

import android.graphics.Bitmap

import com.google.gson.annotations.SerializedName

class BitURL(@field:SerializedName("img")
             internal val img: Bitmap?, @field:SerializedName("url")
             internal val url: String,
             @field:SerializedName("post")
             internal val postLink: String) {
    @SerializedName("gif")
    private var containsGif = false

    init {
        if (img == null) {
            containsGif = true
        }
    }

    fun hasGif(): Boolean {
        return containsGif
    }

    fun setGif(gif: Boolean) {
        containsGif = gif
    }

    fun getUrl(): String {
        return url
    }

    fun getImg(): Bitmap? {
        return img
    }
}
